ctk/common/src/main/kotlin/org/codice/compliance/verification/core/SignatureAndEncryptionVerifier.kt

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compliance.verification.core
+
+import org.apache.cxf.rs.security.saml.sso.SSOConstants
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLCore_5_4_2_b
+import org.codice.compliance.SAMLCore_5_4_2_b1
+import org.codice.compliance.SAMLCore_6_1_b
+import org.codice.compliance.allChildren
+import org.codice.compliance.children
+import org.codice.compliance.utils.TestCommon
+import org.w3c.dom.Node
+
+class SignatureAndEncryptionVerifier(private val node: Node) {
+    companion object {
+        private const val ENCRYPTED_DATA = "EncryptedData"
+    }
+
+    /**
+     * 5 SAML and XML Signature Syntax and Processing
+     * 6 SAML and XML Encryption Syntax and Processing
+     */
+    fun verify() {
+        // Signing
+        verifySignatureSyntaxAndProcessing()
+
+        // Encryption
+        verifyGeneralConsiderations()
+    }
+
+    /** 5.4.2 References */
+    private fun verifySignatureSyntaxAndProcessing() {
+        node.children(SSOConstants.SIGNATURE).forEach {
+            val references = it.allChildren("Reference")
+            if (references.size != 1)
+                throw SAMLComplianceException.create(SAMLCore_5_4_2_b1,
+                        message = "A signature needs to have exactly one Reference, " +
+                                "${references.size} found.",
+                        node = node)
+
+            val uriValue = references[0].attributes.getNamedItem("URI")?.textContent
+                    ?: throw SAMLComplianceException.create(SAMLCore_5_4_2_b1,
+                            message = "URI attribute not found.",
+                            node = node)
+
+            val formattedId = "#" + it.parentNode.attributes?.getNamedItem("ID")?.textContent
+            if (uriValue != formattedId)
+                throw SAMLComplianceException.createWithPropertyMessage(SAMLCore_5_4_2_b,
+                        property = "URI",
+                        actual = uriValue,
+                        expected = formattedId,
+                        node = node)
+        }
+    }
+
+    /** 6.1 - General Considerations **/
+    private fun verifyGeneralConsiderations() {
+        val elements = mutableListOf<Node>()
```

Comment:
✏️ Instead of a mutable list with three calls to `addAll`, just initialize inline:
```kotlin
        val elements = node.children("EncryptedAssertion") + 
                node.children("EncryptedID") + 
                node.children("EncryptedAttribute")
```